client_vm_create.go

```
@@ -2,15 +2,319 @@ package ovirtclient
+	}
+	return nil
+}
+
+// handleAutoPinning updates the VM after creation to set the auto pinning policy configuration.
+func (v *vmCreator) handleAutoPinning() error {
+	optimizeCPUSettings := v.autoPinningPolicy == ovirtsdk.AUTOPINNINGPOLICY_ADJUST
+	_, err := v.ovirtClient.conn.SystemService().VmsService().VmService(v.vm.MustId()).
+		AutoPinCpuAndNumaNodes().
+		OptimizeCpuSettings(optimizeCPUSettings).Send()
+	if err != nil {
+		return errors.Errorf("failed to set the auto pinning policy on the VM!, %v", err)
+	}
+	return nil
+}
+
+func (v *vmCreator) handleVMHugepages() error {
+	customProp, err := ovirtsdk.NewCustomPropertyBuilder().
+		Name("hugepages").
+		Value(fmt.Sprint(v.hugepages)).
+		Build()
+	if err != nil {
+		return errors.Wrap(err, "error setting hugepages custom property")
+	}
+	v.vmBuilder.CustomPropertiesOfAny(customProp)
+
+	return nil
+}
+
+// handleNics replaces the NICs of the VM with nics from specified VNic profiles
+func (v *vmCreator) handleNics() error {
+	vmService := v.ovirtClient.conn.SystemService().VmsService().VmService(v.vm.MustId())
+	nicList, err := vmService.NicsService().List().Send()
+	if err != nil {
+		return errors.Wrap(err, "failed fetching VM network interfaces")
+	}
+
+	// remove all existing nics
+	for _, n := range nicList.MustNics().Slice() {
+		_, err := vmService.NicsService().NicService(n.MustId()).Remove().Send()
+		if err != nil {
+			return errors.Wrap(err, "failed clearing all interfaces before populating new ones")
+		}
+	}
+
+	// re-add nics
+	for i, vNICsID := range v.vNICsIDs {
+		_, err := vmService.NicsService().Add().Nic(
+			ovirtsdk.NewNicBuilder().
+				Name(fmt.Sprintf("nic%d", i+1)).
+				VnicProfileBuilder(ovirtsdk.NewVnicProfileBuilder().Id(vNICsID)).
+				MustBuild()).
+			Send()
+		if err != nil {
+			return errors.Wrap(err, "failed to create network interface")
+		}
+	}
+	return nil
+}
+
+func mapAutoPinningPolicy(policy string) ovirtsdk.AutoPinningPolicy {
+	switch policy {
```

It seems to me that we should only allow correct values here and fail if an invalid value is provided. Also, we should create consts for the enum values instead of hardcoding them as strings.